### ISSUE TITLE
Docker machine hangs on startup. (ubuntu 16.04)

### ISSUE THREAD

**Issue by spoon99** [docker/docker#25040]
**Output of `docker version`:**

```
docker version commands hang.
So I checked docker version with following command.

/usr/bin/docker -v
Docker version 1.11.2, build b9f10c9
```

**Output of `docker info`:**

```
docker info command hang.
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
AWS, Ubuntu 16.04

```
root@ip-10-213-*****~# uname -ar
Linux ip-10-213-***** 4.4.0-28-generic #47-Ubuntu SMP Fri Jun 24 10:09:13 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

Sometime docker engine doesn't startup. It saids activating when I check systemctl status docker.

```
systemctl status docker
● docker.service - Docker Service
   Loaded: loaded (/etc/systemd/system/docker.service; enabled; vendor preset: enabled)
   Active: activating (start) since Tue 2016-07-26 11:15:03 KST; 19min ago
 Main PID: 1191 (docker)
    Tasks: 7 (limit: 512)
   Memory: 40.4M
      CPU: 89ms
   CGroup: /system.slice/docker.service
           └─1191 /usr/bin/docker daemon -H fd://

Jul 26 11:15:03 ip-10-210-****** systemd[1]: Starting Docker Service...
Jul 26 11:15:10 ip-10-213-****** docker[1191]: time="2016-07-26T11:15:10.021767806+09:00" level=info msg="New containerd process, pid: 1376\n"
Jul 26 11:15:16 ip-10-213-****** docker[1191]: time="2016-07-26T11:15:16.148458143+09:00" level=info msg="previous instance of containerd still alive (1376)"
```

When I stop docker and restart with systemctl It works fine.
```
systemctl stop docker
root@ip-10-213-******:~# systemctl start docker
root@ip-10-213-******:~# systemctl status docker
● docker.service - Docker Service
   Loaded: loaded (/etc/systemd/system/docker.service; enabled; vendor preset: enabled)
   Active: active (running) since Tue 2016-07-26 11:37:17 KST; 2s ago
 Main PID: 6428 (docker)
    Tasks: 20 (limit: 512)
   Memory: 101.1M
      CPU: 3.186s
   CGroup: /system.slice/docker.service
           ├─6428 /usr/bin/docker daemon -H fd://
           ├─6437 docker-containerd -l /var/run/docker/libcontainerd/docker-containerd.sock --runtime docker-runc --start-timeout 2m
           └─6651 docker-containerd-shim 833fa3816a9e59261ab5d01d95fd9c057f3f7bd9ccc4732aff1fe5071f6bab27 /var/run/docker/libcontainerd/833fa3816a9e59261ab5d01d

Jul 26 11:37:17 ip-10-213-****** docker[6428]: time="2016-07-26T11:37:17.230355352+09:00" level=info msg="Firewalld running: false"
Jul 26 11:37:17 ip-10-213-****** docker[6428]: time="2016-07-26T11:37:17.320064458+09:00" level=info msg="Default bridge (docker0) is assigned with an IP addres
Jul 26 11:37:17 ip-10-213-****** docker[6428]: time="2016-07-26T11:37:17.367085984+09:00" level=warning msg="Your kernel does not support swap memory limit."
Jul 26 11:37:17 ip-10-213-****** docker[6428]: time="2016-07-26T11:37:17.367982187+09:00" level=info msg="Loading containers: start."
Jul 26 11:37:17 ip-10-213-****** docker[6428]: time="2016-07-26T11:37:17.441731939+09:00" level=info msg="Loading containers: done."
Jul 26 11:37:17 ip-10-213-****** docker[6428]: time="2016-07-26T11:37:17.441758911+09:00" level=info msg="Daemon has completed initialization"
Jul 26 11:37:17 ip-10-213-****** docker[6428]: time="2016-07-26T11:37:17.441778719+09:00" level=info msg="Docker daemon" commit=b9f10c9 graphdriver=overlay vers
Jul 26 11:37:17 ip-10-213-****** systemd[1]: Started Docker Service.
Jul 26 11:37:17 ip-10-213-****** docker[6428]: time="2016-07-26T11:37:17.461922840+09:00" level=info msg="API listen on /var/run/docker.sock"
Jul 26 11:37:17 ip-10-213-****** systemd[1]: Started Docker Service.
```

Any clue for this?

**Comment by AkihiroSuda** [comment]
`killall -SIGUSR1 docker` will show stack traces in journalctl

**Comment by thaJeztah** [comment]
@mlaventure is the "previous instance of containerd still alive (1376)" perhaps related here?

**Comment by mlaventure** [comment]
Possible. I'd love to see the stacktrace though.

@spoon99 could you do a `kill -USR1  $(cat /run/docker.pid)` as suggested by @AkihiroSuda and give us the output as found in the daemon logs?

**Comment by spoon99** [comment]
Sorry for late response. It was hard to reproduce.

Today I got a instance, which had same problem.

I tried `kill -USR1 $(cat /run/docker.pid)` and nothing was showed on `journalctl -f`.

```
root@ip-10-***********:~# killall -SIGUSR1 docker
root@ip-10-***********:~# kill -USR1 $(cat /run/docker.pid)
root@ip-10-***********:~# kill -USR1 $(cat /run/docker.pid)
root@ip-10-***********:~# killall -SIGUSR1 docker
root@ip-10-***********:~# ps -ef | grep docker
root      1194     1  0 17:24 ?        00:00:00 /usr/bin/docker daemon -H fd://
coupang   7374  7358  0 17:36 pts/0    00:00:00 docker ps
root      7552  7526  0 17:41 pts/3    00:00:00 grep --color=auto docker
root@ip-***********:~# kill -USR1 1194

```


```
root@ip-10-213-64-177:~# journalctl -f -u docker
-- Logs begin at Thu 2016-08-18 17:23:56 KST. --
Aug 18 17:24:28 ip-10-213-64-177 systemd[1]: Starting Docker Service...
Aug 18 17:24:34 ip-10-213-64-177 docker[1194]: time="2016-08-18T17:24:34.818000495+09:00" level=info msg="New containerd process, pid: 1377\n"
Aug 18 17:24:40 ip-10-213-64-177 docker[1194]: time="2016-08-18T17:24:40.822328029+09:00" level=info msg="previous instance of containerd still alive (1377)"
```

**Comment by mlaventure** [comment]
@spoon99 your issue seems to be the same as https://github.com/docker/docker/issues/25292

It is fixed in `1.12.0`

**Comment by spoon99** [comment]
OK, Thanks I'll try upgrading docker engine to 1.12.0
Thanks,
